osquery/tables/system/windows/windows_search.cpp

```diff
@@ -0,0 +1,244 @@
+
+  std::wstringstream wss;
+
+  DBSTATUS status;
+  cCommand.GetStatus(columnIndex, &status);
+  if (status == DBSTATUS_S_ISNULL) {
+    wss << "NULL";
+  } else if (status == DBSTATUS_S_OK || status == DBSTATUS_S_TRUNCATED) {
+    DBTYPE type;
+    cCommand.GetColumnType(columnIndex, &type);
+    switch (type) {
+    case DBTYPE_VARIANT:
+      writePropVariant(
+          *(static_cast<PROPVARIANT*>(cCommand.GetValue(columnIndex))), wss);
+      break;
+    case DBTYPE_WSTR: {
+      DBLENGTH cbLen;
+      cCommand.GetLength(columnIndex, &cbLen);
+      WCHAR szBuffer[2048];
+      StringCchCopyN(szBuffer,
+                     ARRAYSIZE(szBuffer),
+                     static_cast<WCHAR*>(cCommand.GetValue(columnIndex)),
+                     cbLen / sizeof(WCHAR));
+      wss << szBuffer;
+    } break;
+    case DBTYPE_I1:
+      wss << *static_cast<UCHAR*>(cCommand.GetValue(columnIndex));
+      break;
+    case DBTYPE_UI2:
+      wss << *static_cast<USHORT*>(cCommand.GetValue(columnIndex));
+      break;
+    case DBTYPE_I2:
+      wss << *static_cast<SHORT*>(cCommand.GetValue(columnIndex));
+      break;
+    case DBTYPE_UI4:
+      wss << *static_cast<DWORD*>(cCommand.GetValue(columnIndex));
+      break;
+    case DBTYPE_I4:
+      wss << *static_cast<INT*>(cCommand.GetValue(columnIndex));
+      break;
+    case DBTYPE_UI8:
+      wss << *static_cast<ULONGLONG*>(cCommand.GetValue(columnIndex));
+      break;
+    case DBTYPE_I8:
+      wss << *static_cast<LONGLONG*>(cCommand.GetValue(columnIndex));
+      break;
+    case DBTYPE_DATE:
+      wss << dateToUnixTime(
+          *static_cast<DATE*>(cCommand.GetValue(columnIndex)));
+      break;
+    default:
+      wss << "unhandled database type " << type;
+      break;
+    }
+  } else {
+    wss << "error reading column";
+  }
+
+  std::wstring wideStr = wss.str();
+  std::wstring_convert<std::codecvt_utf8<wchar_t>> converter;
+  return converter.to_bytes(wideStr);
+}
+
+std::vector<std::map<std::string, std::string>> executeWindowsSearchQuery(
+    CSession& cSession, const std::string& query) {
+  HRESULT hr;
+  std::vector<std::map<std::string, std::string>> results;
+
+  CCommand<CDynamicAccessor, CRowset> cCommand;
+  hr = cCommand.Open(cSession, query.c_str());
+
+  if (!SUCCEEDED(hr)) {
+    LOG(ERROR) << "error executing query";
+    return results;
+  }
+
+  for (hr = cCommand.MoveFirst(); S_OK == hr; hr = cCommand.MoveNext()) {
+    std::map<std::string, std::string> row;
+
+    for (DBORDINAL i = 1; i <= cCommand.GetColumnCount(); i++) {
+      auto columnName = cCommand.GetColumnName(i);
+      std::string columnNameStr = wstringToString(columnName);
+
+      // Add the column name and value to the row map.
+      row[columnNameStr] = ccomandColumnStringValue(cCommand, i);
+    }
+
+    results.push_back(row);
+  }
+
+  cCommand.Close();
+  return results;
+}
+
+QueryData genWindowsSearch(QueryContext& context) {
+  QueryData results;
+
+  HRESULT hr = CoInitialize(NULL);
```

Comment:
You can use CoInitializeEx(0, COINIT_MULTITHREADED); and not check the return value as COM is most likely already initialized.